src/mixed-state-qvm.lisp

```diff
@@ -0,0 +1,135 @@
+;;;; src/mixed-state-qvm.lisp
+;;;;
+;;;; Authors: Robert Smith
+;;;;          Erik Davis
+;;;;          Sophia Ponte
+
+(in-package #:qvm)
+
+;;; This file implements a qvm that evolves pure and mixed states. 
+
+;;; General Overview
+
+;;; The MIXED-STATE-QVM is an implementation of a QVM that can evolve
+;;; either a pure or mixed state using a DENSITY-MATRIX-STATE. The
+;;; MIXED-STATE-QVM inherits most of its core behavior from BASE-QVM,
+;;; but provides more specialized functionality for methods like
+;;; TRANSITION, MEASURE, and MEASURE-ALL. Since the MIXED-STATE-QVM
+;;; uses a DENSITY-MATRIX-STATE, it converts quil gates to
+;;; superoperators for gate application. Additional custom
+;;; superoperators can be defined in the SUPEROPERATOR-DEFINITIONS of
+;;; the MIXED-STATE-QVM.
+
+(defclass mixed-state-qvm (base-qvm)
+  ((state :accessor state
+          :initarg :state
+          :type (or null density-matrix-state)))
+  (:documentation "A qvm for dealing with mixed-state simulations."))
+
+(defmethod temporary-state ((qvm mixed-state-qvm))
```

Comment:
Is this (and the `setf` method below) used anywhere?

Reply:
I actually don't think so -- I included those methods while I was refactoring since I got rid of the `temporary-state` accessor in the `density-qvm`.